### PR TITLE
feat(system-tests): Add simple injection tests for the installer

### PR DESCRIPTION
**Pull request by BaptisteFoy**
## Motivation
We'd like to test the auto injection scenario when everything is installed through the installer
See pipeline for https://github.com/DataDog/dd-trace-java/pull/6912 to see that it works :)

## Changes
Creates a new scenario `SIMPLE_INSTALLER_AUTO_INJECTION`

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: utils/_context/_scenarios.py [hunk 1]
```diff
@@ -1695,6 +1711,10 @@ class scenarios:
         "K8S_LIB_INJECTION_FULL", doc=" Kubernetes Instrumentation complete scenario"
     )
 
+    installer_auto_injection = InstallerAutoInjectionScenario(
+        "INSTALLER_AUTO_INJECTION", doc="Installer auto injection scenario (minimal test scenario)"
```
Comment (BaptisteFoy) [inline utils/_context/_scenarios.py:1715]: I would've loved to call it `SIMPLE_INSTALLER_AUTO_INJECTION` but we're hitting the max AMI name size (128 chars) 😅

File: utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml [hunk 2]
```diff
@@ -0,0 +1,182 @@
+# Installs the installer package
+- os_type: linux
+  copy_files:
+    - name: copy-tracer-debug-config
+      local_path: utils/build/virtual_machine/provisions/auto-inject/tracer_debug/debug_config.yaml
+    - name: copy-docker-config
+      local_path: utils/build/virtual_machine/provisions/auto-inject/docker/docker_config.yaml
+
+  # TODO(baptiste): use install script
```
Comment (BaptisteFoy) [inline utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml:9]: I'll handle that in another PR once we have a working install script

File: utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        yum_url="yumtesting.datad0g.com/suse/testing"
+        yum_repo_version="${DD_PIPELINE_ID}-i7/7"
+
+        RPM_GPG_KEYS=("DATADOG_RPM_KEY_CURRENT.public" "DATADOG_RPM_KEY_B01082D3.public" "DATADOG_RPM_KEY_FD4BF915.public" "DATADOG_RPM_KEY_E09422B3.public")
+        separator='\n       '
+        for key_path in "${RPM_GPG_KEYS[@]}"; do
+            gpgkeys="${gpgkeys:+"${gpgkeys}${separator}"}https://${keys_url}/${key_path}"
+        done
+        $sudo_cmd sh -c "echo -e '[datadog]\nname = Datadog, Inc.\nbaseurl = https://${yum_url}/${yum_repo_version}/${ARCH}/\nenabled=1\ngpgcheck=1\nrepo_gpgcheck=1\npriority=1\ngpgkey=${gpgkeys}' > /etc/zypp/repos.d/datadog.repo"
+        $sudo_cmd zypper -n --gpg-auto-import-keys refresh
```
Comment (raphaelgavache) [inline utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml:148]: you can replace above lines by running  
DD_REPO_URL=datad0g.com DD_AGENT_DIST_CHANNEL=beta DD_NO_AGENT_INSTALL=true bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script_agent7.sh)"

File: utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+    sudo chown -R dd-agent:dd-agent /opt/datadog-installer
+    sudo chmod -R 777 /opt/datadog-packages
+
+    sudo mkdir -p /etc/datadog-agent/inject
+    sudo cp docker_config.yaml /etc/datadog-agent/inject/docker_config.yaml
+    sudo cp debug_config.yaml /etc/datadog-agent/inject/debug_config.yaml
+
+
+  # remote-command: |
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml:175]: Yes that's the goal. I can remove the commented lines if you prefer

File: binaries/README.md
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply binaries/README.md:1]: Oops, yes, that was a test leftover. Good catch!

File: utils/build/virtual_machine/provisions/auto-inject/auto-inject-tested_components.yml [hunk 5]
```diff
@@ -1,9 +1,24 @@
...
 - os_type: linux
-  os_distro: rpm
   remote-command: |
-      echo "{'host':'$(hostname -I)','agent':'$(rpm -qa --queryformat '%{VERSION}-%{RELEASE}' datadog-agent)','datadog-apm-inject':'$(rpm -qa --queryformat '%{VERSION}-%{RELEASE}' datadog-apm-inject)','datadog-apm-library-$DD_LANG': '$(rpm -qa --queryformat '%{VERSION}-%{RELEASE}' datadog-apm-library-$DD_LANG)','docker':'$(docker -v || true)'}"
-- os_type: linux
-  os_distro: deb
-  remote-command: |
-      version_agent=$((dpkg -s datadog-agent || true)  | grep Version  | head -n 1 )&& version=$((dpkg -s datadog-apm-inject || true) | grep Version) && version_lang=$(dpkg -s datadog-apm-library-$DD_LANG| grep Version) && echo "{'host':'$(hostname -I)','agent':'${version_agent//'Version:'/}','datadog-apm-inject':'${version//'Version:'/}','datadog-apm-library-$DD_LANG':'${version_lang//'Version:'/}','docker':'$(docker -v || true)'}"
\ No newline at end of file
+      export agent_path="$(readlink -f /opt/datadog-packages/datadog-agent/stable)"
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply utils/build/virtual_machine/provisions/auto-inject/auto-inject-tested_components.yml]: I thought I created another file for this, my bad. I moved it to another file in [31d6661](https://github.com/DataDog/system-tests/pull/2360/commits/31d6661b4933621e3266de77f3c13c771976a456)

File: utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (robertomonteromiguel) [reply utils/build/virtual_machine/provisions/installer-auto-inject/auto-inject_installer_script.yml:176]: We don't need to remove these lines.....

File: utils/_features.py [hunk 6]
```diff
@@ -2106,3 +2106,13 @@ def debugger_pii_redaction(test_object):
         """
         pytest.mark.features(feature_id=291)(test_object)
         return test_object
+
+    @staticmethod
+    def installer_auto_instrumentation(test_object):
+        """
+        Installer auto-instrumentation
+
+        https://feature-parity.us1.prod.dog/#/?feature=292
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply utils/_features.py:2125]: Done, thanks again for the help: https://feature-parity.us1.prod.dog/#/?page=5&language=9&feature=292

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**